src/Simplex/Messaging/Agent/Env/SQLite.hs

```diff
@@ -86,8 +86,9 @@ newSMPAgentEnv cfg = do
   store <- liftIO $ createSQLiteStore (dbFile cfg) (dbPoolSize cfg) Migrations.app
   clientCounter <- newTVarIO 0
   randomServer <- newTVarIO =<< liftIO newStdGen
-  agentCredential <- liftIO $ loadServerCredential (agentPrivateKeyFile cfg) (agentCertificateFile cfg)
-  return Env {config = cfg, store, idsDrg, clientCounter, reservedMsgSize, randomServer, agentCredential}
+  serverCredential <- liftIO $ loadServerCredential (agentPrivateKeyFile cfg) (agentCertificateFile cfg)
+  let tlsServerParams = mkTLSServerParams serverCredential
```

Comment:
it makes sense to compose loadServerCredential and mkTLSServerParams into one function loadTLSServerParams that accepts key file and cert file paths (or getTLSServerParams) in Transport and call it from both Agent and Server

Reply:
started doing it but liked it less..